v2/lint/source.go

```diff
@@ -25,17 +25,18 @@ import (
 type LintSource string
 
 const (
-	UnknownLintSource        LintSource = "Unknown"
-	RFC5280                  LintSource = "RFC5280"
-	RFC5480                  LintSource = "RFC5480"
-	RFC5891                  LintSource = "RFC5891"
-	CABFBaselineRequirements LintSource = "CABF_BR"
-	CABFEVGuidelines         LintSource = "CABF_EV"
-	MozillaRootStorePolicy   LintSource = "Mozilla"
-	AppleCTPolicy            LintSource = "Apple"
-	ZLint                    LintSource = "ZLint"
-	AWSLabs                  LintSource = "AWSLabs"
-	EtsiEsi                  LintSource = "ETSI_ESI"
+	UnknownLintSource                   LintSource = "Unknown"
+	RFC5280                             LintSource = "RFC5280"
+	RFC5480                             LintSource = "RFC5480"
+	RFC5891                             LintSource = "RFC5891"
+	CABFBaselineRequirementsOverbearing LintSource = "CABF_BR_Overbearing"
```

Comment:
I did comment on this somewhere when I introduced, but I can't find the comment anymore. This PR gets confusing. The background of this additional set of LintSources is, that we do have rules, which come from the CAB BRGs that do apply for every certificate which is under an TLS capable CA, even if this certificate itself is not a TLS certificate. Now we had two options to represent this in the LINTS: option 1 was to update all other lints to reflect this (what I did at the beginning but got negative feedback here) and option 2 is to introduce this new lint source. At least I didn't have any more ideas how to handle the situation. If you have something else in your mind, please feel free to propose a different implementation and I'll update the sources accordingly.

Reply:
> This PR gets confusing.

Definitely, sorry for the thread maze we made here 😆 

> Now we had two options to represent this in the LINTS: option 1 was to update all other lints to reflect this (what I did at the beginning but got negative feedback here) and option 2 is to introduce this new lint source. At least I didn't have any more ideas how to handle the situation.

Hmm, I see. Given that `w_ocsp_id_pkix_ocsp_nocheck_ext_not_included_non_server_auth` is conservatively returning a `Warning` level finding I'm not sure I understand why we can't place these two new lints under the existing CABF source. What am I missing?

I'm inclined to stamp this even with the new source if others don't feel strongly. It would be possible to revisit in the future.

Reply:
@cpu If we list `w_ocsp_id_pkix_ocsp_nocheck_ext_not_included_non_server_auth` in `CABFBaselineRequirements` then we have to change line 86 of base.go to allow checks for non-server-auth-certificates for lints of the source `CABFBaselineRequirements`. But if we do this, we get plenty of new warnings and errors which are false positives.

Reply:
@cpu Do you think I would get this PR faster through the door if I would split it into two PRs? One for errors and one for warnings?

Reply:
@RufusJWB 

I think one of the challenges I've had is understanding the design constraints for your decisions, and I think that makes it a bit harder to try to reverse engineer to figure out what we can do better here. I think this is where, in thinking about the future, we might want to iterate and explore options on a GH issue first, just to make sure we capture the constraints (like you've done in https://github.com/zmap/zlint/pull/472/files#r508462206 )

I mention this, because
> But if we do this, we get plenty of new warnings and errors which are false positives.

I haven't really seen the breakdown here to understand what is meant here. I thought we'd found a viable path in previous iterations, which was including OCSP for the scope of the BRGs. I apologies that both personal and professional life have gotten busy here, but could you remind me why this approach doesn't work? I believe it was because you thought some OCSP responders would end up violating certain requirements, right? Could you remind me what those are?

I think we're at a place where there's fundamentally not any 'great' answer here, and the question we've been examining is whether to over-warn or under-warn. There was some discussion with how CAs are integrating, and I thought we'd accepted that over-warning was acceptable here. It sounds like you don't believe that's the case, so I'm trying to understand more.

Reply:
@sleevi I'm really sorry, that this PR became such a big mess. I'm for sure not happy neither since I invested also a lot of my time to over and over update this PR. I think at the core the problem was, that I did try to cover both "warning" and "error" in one PR. This made things complex, especially because the "warning" lint is covering certificates which usually don't fall under the BRGs. Furthermore, I did try to follow all the proposals given by the various contributors and some advice I found in the discussion of other bugs. Wherever possible I did try to document this either in the commit message or in the discussion here in this PR. But anyhow, the situation has room for improvement and I have learned a lot about working in GitHub. Based on the proposal of @cardonator I closed this PR and created two new PRs, one for TLS certificates (#490 ) and one for non-TLS certificates (#491 ). Would you never the less be willing to review also the two new PRs? I think with two new PRs we are very close to getting this issue solved.